### PR TITLE
More block import optimizations

### PR DESCRIPTION
**Pull request by arkpar**

### CODE REVIEW

File: libethereum/BlockQueue.cpp [hunk 1]
```diff
@@ -38,7 +38,8 @@ const char* BlockQueueChannel::name() { return EthOrange "▣┅▶"; }
 
 BlockQueue::BlockQueue()
 {
-	for (unsigned i = 0; i < thread::hardware_concurrency(); ++i)
+	unsigned verifierThreads = std::max(thread::hardware_concurrency(), 3U) - 2U;
```
Comment (chfast) [inline libethereum/BlockQueue.cpp]: Explanation would be nice.

**Comment by chfast** [comment]
Looks Good To Me™, but can you explain what is this optimization about?

File: libethcore/EthashAux.cpp [hunk 2]
```diff
@@ -101,13 +101,13 @@ uint64_t EthashAux::number(h256 const& _seedHash)
...
 {
-	RecursiveGuard l(x_lights);
+	WriteGuard l(x_lights);
 	m_lights.erase(_s);
 }
 
 EthashAux::LightType EthashAux::light(h256 const& _seedHash)
 {
-	RecursiveGuard l(get()->x_lights);
+	ReadGuard l(get()->x_lights);
```
Comment (arkpar) [inline libethcore/EthashAux.cpp]: Prevents contention on parallel ethash verification

File: libethereum/BlockChain.cpp [hunk 3]
```diff
@@ -317,7 +317,7 @@ tuple<h256s, h256s, bool> BlockChain::sync(BlockQueue& _bq, OverlayDB const& _st
 	{
 		try
 		{
-			auto r = import(block.first, block.second, _stateDB);
+			auto r = import(block.first, block.second, _stateDB, ImportRequirements::Default & ~ImportRequirements::ValidNonce);
```
Comment (arkpar) [inline libethereum/BlockChain.cpp]: Nonce is already checked by verifier thread at this point. No need to check it again

Comment (chfast) [inline libethereum/BlockChain.cpp]: This comment should be in the code.

**Pull request merged**